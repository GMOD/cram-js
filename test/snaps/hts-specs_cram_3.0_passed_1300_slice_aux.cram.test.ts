import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/1300_slice_aux.cram')
