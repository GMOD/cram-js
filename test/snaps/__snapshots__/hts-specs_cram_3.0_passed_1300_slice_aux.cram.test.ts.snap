// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/1300_slice_aux.cram without error 1`] = `
[
  {
    "fileId": "_.cram",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 3348343310,
      "landmarks": [
        0,
        163,
      ],
      "length": 253,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 3348343310,
        "landmarks": [
          0,
          163,
        ],
        "length": 253,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/cram/passed/../ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/cram/passed/../ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              40,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1000,
            ],
          },
          "parametersBytes": 5,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -300,
            ],
          },
          "parametersBytes": 8,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 319,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 1622469697,
      "landmarks": [
        164,
      ],
      "length": 508,
      "numBases": 200,
      "numBlocks": 8,
      "numLandmarks": 1,
      "numRecords": 2,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 319,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 1622469697,
        "landmarks": [
          164,
        ],
        "length": 508,
        "numBases": 200,
        "numBlocks": 8,
        "numLandmarks": 1,
        "numRecords": 2,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "match",
              "sequenceId": 0,
              "uniqueId": 490,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 489,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "match",
              "sequenceId": 0,
              "uniqueId": 489,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 490,
          },
        ],
        "header": {
          "_endPosition": 551,
          "_size": 63,
          "compressedSize": 59,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            2,
            0,
            6,
            5,
            11,
            12,
            15,
            16,
            17,
            255,
            255,
            255,
            255,
            15,
            188,
            14,
            187,
            152,
            12,
            130,
            56,
            146,
            25,
            54,
            223,
            160,
            201,
            234,
            161,
            96,
            66,
            68,
            66,
            99,
            4,
            0,
            0,
            0,
            219,
            142,
            125,
            238,
            83,
            68,
            66,
            99,
            4,
            0,
            0,
            0,
            120,
            146,
            7,
            0,
          ],
          "contentId": 0,
          "contentPosition": 488,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 3448572300,
          "parsedContent": {
            "_endPosition": 354,
            "_size": 35,
            "contentIds": [
              11,
              12,
              15,
              16,
              17,
            ],
            "md5": [
              188,
              14,
              187,
              152,
              12,
              130,
              56,
              146,
              25,
              54,
              223,
              160,
              201,
              234,
              161,
              96,
            ],
            "numBlocks": 6,
            "numContentIds": 5,
            "numRecords": 2,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 59,
        },
      },
    ],
  },
]
`;
